### ISSUE TITLE
Update single package broken

### ISSUE THREAD

**Issue by tschechniker** [composer/composer#4770]
Form my side the function for update a single package is broken. If i run composer update <package> also other packages will be updated. 

For example i require two packages (A and B) I have installed the Version 1.0.0 of both packages:

````
require:{
    "A":"~1.0",
    "B":"~1.0"
}
````
Now both will get updated to Version 1.0.1. If i run 
```bash
composer update A
````
package B will also be updated to Version 1.0.1. A and B don't have any relationships between each other. 

From my side it seems like the argument is getting ignored. Behaviour is the same as running update without the specific package.

**Comment by alcohol** [comment]
Can you show us some actual composer.json contents and what command you were running? These "dummy" examples don't provide anything of value.

**Comment by tschechniker** [comment]
I did some more research. This could be just a bad behaviour. Lets say you have the following composer.json:

````json
{
    "require": {
        "symfony/symfony": "~2.8",
        "slim/slim": "~2.6"
    }
}
````

Now you want to update slim:

````bash
composer update slim/slim
````

So far so good. Now if you delete the symfony folder inside the vendor and run composer update slim/slim again it will also reinstall the symfony application. From my side i delete the .git folder inside the vendor to create a unique artefact. Every time i use the artefact i call composer update <specific_package> to get the latest version of this very specific package. Why does composer even care about the other packages if i specify a single one?

**Comment by alcohol** [comment]
I think that has to do with the lock file also. It references a package that is no longer found in the vendor directory. Not sure how or why Composer detects this and automatically resolves it.

**Comment by tschechniker** [comment]
So how can we solve this Issue? Currently from my opinion the update for a single package is broken because of this behaviour.

**Comment by P0rnflake** [comment]
@tschechniker if you delete something from your verdor directory composer update installs it again from your .lock file, because update seems to trigger an implicit install, even if you don't have the package whitelisted for update.

Please correct me if I am wrong.

**Comment by LeeSaferite** [comment]
I see a very similar issue but I'm not sure it's the same or if it needs a new issue.
If I update any single package, all of the packages that use '*' or 'dev-master' also get updated, even with no interdependencies.

**Comment by Seldaek** [comment]
@LeeSaferite that use case works for me, so if you really have an issue please find a way to give us an exact repro case with a full composer.json and instructions. And yes I think I'd open a new issue.

**Comment by LeeSaferite** [comment]
I have the lock files, but cannot share them publicly. How can I get them to you?

**Comment by Seldaek** [comment]
You can email me I guess (see my profile page) but I don't know when I'll have time to look at it.

**Comment by LeeSaferite** [comment]
I'll try to make one I can share first. Thanks.

**Comment by Seldaek** [comment]
Should be fixed by #5210 so closing
